tests/frontier/opcodes/test_swap.py

```diff
@@ -0,0 +1,130 @@
+"""
+A State test for the set of `SWAP*` opcodes.
+Ported from: https://github.com/ethereum/tests/blob/develop/src/GeneralStateTestsFiller/VMTests/vmTests/swapFiller.yml.
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork, Frontier, Homestead
+from ethereum_test_tools import Account, Alloc, Environment
+from ethereum_test_tools import Opcodes as Op
+from ethereum_test_tools import StateTestFiller, Storage, Transaction, Bytecode
+
+
+@pytest.mark.parametrize(
+    "swap_opcode",
+    [getattr(Op, f"SWAP{i}") for i in range(1, 17)],  # Dynamically parametrize SWAP opcodes (SWAP1 to SWAP16)
+    ids=lambda op: str(op),
+)
+@pytest.mark.valid_from("Frontier")
+def test_swap(state_test: StateTestFiller, fork: Fork, pre: Alloc, swap_opcode: Op):
+    """
+    The set of `SWAP*` opcodes swaps the top of the stack with a specific element.
+
+    In this test, we ensure that the set of `SWAP*` opcodes correctly swaps
+    the top element with the nth element and stores the result in storage.
+    """
+    env = Environment()
+
+    # Calculate which position we're swapping with (1-based index)
+    swap_pos = swap_opcode.int() - 0x90 + 1
+
+    # Generate stack values
+    # We'll use position numbers as values to make it clear what's being swapped
+    # For SWAP1: [1, 0]
+    # For SWAP2: [2, 1, 0]
+    # etc.
+    stack_values = list(range(swap_pos + 1))
+
+    # Push the stack values onto the stack (in reverse order).
+    contract_code = Bytecode()
+    for value in reversed(stack_values):
+        contract_code += Op.PUSH1(value)
+
+    # Perform the SWAP operation.
+    contract_code += swap_opcode
+
+    # Store the top of the stack in storage slot 0.
+    contract_code += Op.PUSH1(0) + Op.SSTORE
+
+    # Deploy the contract with the generated bytecode.
+    contract = pre.deploy_contract(contract_code)
```

Comment:
its a little difficult to read the code like this. I prefer when it is at one place 
code = Op.... 
      + Op.
       +Op.

use sum to construct opcodes from loops